In [1]:
import numpy as np


# 1
aragonite - A form of calcium carbonate, less stable than calcite, pteropods make shells out of aragonite (~10% of ocean calcium carbonate)

calcite - A form of calcium carbonate, most stable form at all ocean temperatures and pressures, coccolithophores and foraminifera make shells out of calcite (~90% of ocean calcium carbonate)

calcium carbonate saturation horizon - The ocean depth where $\Delta CO_3^{2-} = 0$ $(\Delta CO_3^{2-} = [CO_3^{2-}] - [CO_3^{2-}]_{sat})$

Rain ratio - The inorganic to organic carbon export ratio ($\frac{\phi_{CaCO_3}}{\phi_{C_{org}}}$)

# 2

In the 2-box model, the rate of change of the surface concentration of Carbon is defined by:

(1) $\frac{dC_s}{dt} = \frac{v}{V_s}(C_d - C_s) - \frac{\phi}{V_s}$

Where $C_s$ is the surface Carbon concentration, $C_d$ is the deep ocean Carbon concentration, $v$ is the surface-deep ocean exchange rate, $V_s$ is the volume of the upper box (surface ocean), and $\phi$ is the carbon export rate.  If we assume steady state, we get:
 
(2) $\frac{dC_s}{dt} = 0 = \frac{v}{V_s}(C_d - C_s) - \frac{\phi}{V_s}$

(3) $\phi = v(C_d - C_s)$

Next, we define potential Alkilinity: $P_{Alk} = Alk + [NO_3^-]$.  Variations in potential Alkilinity are governed only by the formation and dissolution of Calcium Carbonate, not by photosynthesis or remineralization. Using the vertical gradient of potential Alkilinity, we can define the downward export of $CaCO_3$:

(4) $\phi_{CaCO_3} = v*r_{CaCo_3:Alk}*([P_{Alk}]_d - [P_{Alk}]_s)$

Where $r_{CaCo_3:Alk}$ is the ratio of Calcium Carbonate to Alkilinity.  We can also define the downward export of organic carbon in terms of the vertical gradient of Nitrate and the ratio of Carbon to Nitrogen ($r_{C:N}$):

(5) $\phi_{C_{org}} = v*r_{C:N}*([NO_3^-]_d - [NO_3^-]_s)$

The rain ratio is the ratio of inorganic to organic Carbon export, or $\frac{\phi_{CaCO_3}}{\phi_{C_{org}}}$.  Dividing equation (4) by (5) gives the rain ratio:

(6) $\frac{\phi_{CaCO_3}}{\phi_{C_{org}}} = \frac{v*r_{CaCo_3:Alk}*([P_{Alk}]_d - [P_{Alk}]_s)}{v*r_{C:N}*([NO_3^-]_d - [NO_3^-]_s)}$

The ratio of Carbon to Nitrogen ($R_{C:N}$) is determined by Redfield's ratio.  Thus, $R_{C:N}$ = 106/16.  The ratio of $CaCO_3$ to Alkilinity ($r_{CaCo_3:Alk}$) is ... HELP ....  Thus, $r_{CaCo_3:Alk}$ = 1/2. Plugging these numbers into equation (6) and simplifying gives:

(7) $\frac{\phi_{CaCO_3}}{\phi_{C_{org}}} = \frac{([P_{Alk}]_d - [P_{Alk}]_s)}{13.25*([NO_3^-]_d - [NO_3^-]_s)}$

# 3

The primary driver for dissolution of CaCO3 is the state of saturation of the water with respect to the mineral phase: $\Omega = \frac{[CO_3^{2-}][Ca^{2+}]}{K_{sp}^{CaCO_3}}$

# 4

The following equations are used to calculate the saturation states of Calcite and Aragonite:

$\Delta CO_3^{2-} = [CO_3^{2-}]_{obs} - [CO_3^{2-}]_{sat}$

$\Omega = \frac{[CO_3^{2-}]_{obs}}{[CO_3^{2-}]_{sat}}$

where $[CO_3^{2-}]_{obs} = Alk - DIC$

In [4]:
DIC = 2300 #Dissolved inorganic carbon (micro-mol/kg)
Alk = 2400 #Alkilinity (micro-mol/kg)
cal_sat = 124.2 #Saturation concentration of Calcite at depth = 6000dbar, T = 0C, S = 35
arg_sat = 183.7 #Saturation concentration of Aragonite at depth = 6000dbar, T = 0C, S = 35

CO3_obs = Alk - DIC #observed concentration of CO3
delCO3_cal = CO3_obs - cal_sat #Saturation state Delta CO3 for calcite
delCO3_arg = CO3_obs - arg_sat #Saturation state Delta CO3 for aragonite
ohm_cal = CO3_obs/cal_sat #Saturation state omega for calcite
ohm_arg = CO3_obs/arg_sat #Saturation state omega for aragonite

print(delCO3_cal, ohm_cal)
print(delCO3_arg, ohm_arg)

-24.200000000000003 0.8051529790660226
-83.69999999999999 0.5443658138268918


For calcite, $\Delta CO_3^{2-}$ = -24.2 micro-mol/kg and $\Omega$ = 0.805

For aragonite, $\Delta CO_3^{2-}$ = -83.7 micro-mol/kg and $\Omega$ = 0.544

Both calcite and aragonite are undersaturated.